src/dz-4/hw4.ts

```diff
@@ -0,0 +1,79 @@
+// 1
+interface Calculator {
+    add(a: number, b: number): number;
+    subtract(a: number, b: number): number;
+    multiply(a: number, b: number): number;
+    division(a: number, b: number): number;
+
+}
+
+const calculate = (calculator: Calculator, operation: string, a: number, b: number): number => {
+    switch (operation) {
+        case 'add':
+            return calculator.add(a, b);
+        case 'subtract':
+            return calculator.subtract(a, b);
+        case 'multiply':
+            return calculator.multiply(a, b);
+        case 'division':
```

Comment:
You can create an enum data structure to handle those strings. It's a great way to keep your code organized and easy to read. Let me know if you need any help with that:-)